### PR TITLE
Unbundle parts of the Bazel configuration for pyiree from IREE

### PR DESCRIPTION
**Pull request by marbre**
At the moment the Bazel build configuration for `IREE` and `pyiree` is spread into multiple files, which makes it hard to follow and nearly impossible to split the the two into different repositories. As discussed on the [mailinglist](https://groups.google.com/forum/#!topic/iree-discuss/V1IJD9IZs-M), the Bazel configuration should be refactored. This PR introduces first patches to unbundle the build configuration:

* Moves most of the python relevant parts from `iree/build_defs.bzl` to `bindings/python/build_defs.bzl` and `bindings/python/pyiree/build_defs.bzl`
* Moves the files in `build_tools/python` to `bindings/python/build_tools/python`
* Fixes an typo in `bindings/python/pyiree/build_defs.bzl`:
  Deps were defined as `PYTHON_HEADER_DEPS`, but Bazel configuration files in subdirectories loaded `PYTHON_HEADERS_DEPS` (HEADER**S** instead of HEADER)

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
FYI it might be difficult to do this unbundling properly as an external contributor because internally some of these are shims over the internal implementations. We can probably take this PR and fix it up internally though

**Comment by stellaraccident** [comment]
Thanks for the cleanups. As Geoffrey says, we need to make some simultaneous upstream fixes, so submission will be manual on our end (and you may see some additional changes that we end up having to make in the process).

**Comment by marbre** [comment]
Sure, feel free do adopt or cherry-pick the changes you can make use of internally :) Just tripped over this on my way to understand what Bazel is actually doing.

**Comment by stellaraccident** [comment]
I'm giving this a shot at integrating now...
